test/tree-sitter-corpus.jl.faceup

```diff
@@ -0,0 +1,784 @@
+
+a[«c:1», «c:2», «c:3»]
+a[«c:1», :]
+«s:"foo"»[«c:1»]
+
+«x:# ==============================»
+«x:# type parametrized expressions»
+«x:# ==============================»
+
+«t:Vector»{«t:Int»}
+«t:Vector»{<:«t:Number»}
+«:julia-ts-interpolation-expression-face:$(»«D:usertype»«:julia-ts-interpolation-expression-face:)»{«t:T»}
+
+{«:julia-ts-quoted-symbol-face::x»} ~ normal(«c:0», «c:1»)
+
+«x:# ==============================»
+«x:# function call expressions»
+«x:# ==============================»
+
+f()
+g(«s:"hi"», «c:2»)
+h(d...)
+
+f(e; «:julia-ts-quoted-symbol-face:f» = g)
+g(arg; kwarg)
+
+«k:new»{typeof(xs)}(xs)
+
+«x:# ========================================»
+«x:# function call expressions with do blocks»
+«x:# ========================================»
+
+reduce(xs) «k:do» x, y
+  f(x, y)
+«k:end»
+
+«x:# ==============================»
+«x:# macro call expressions»
+«x:# ==============================»
+
+«:julia-ts-macro-face:@assert» x == y «s:"a message"»
+
+«:julia-ts-macro-face:@testset» «s:"a"» «k:begin»
+  «v:b» = c
+«k:end»
+
+«:julia-ts-macro-face:@.» a * x + b
+
+joinpath(«:julia-ts-macro-face:@__DIR__», «s:"grammar.js"»)
+
+«:julia-ts-macro-face:@macroexpand» «:julia-ts-macro-face:@async» accept(socket)
+
+«:julia-ts-macro-face:@Meta.dump» «c:1», «c:2»
+Meta.«:julia-ts-macro-face:@dump» «v:x» = «c:1»
+
+«x:# ==============================»
+«x:# closed macro call expressions»
+«x:# ==============================»
+
+«:julia-ts-macro-face:@enum»(Light, red, yellow, green)
+«f:f»(«:julia-ts-macro-face:@nospecialize»(x)) = x
+
+«:julia-ts-macro-face:@m»[«c:1», «c:2»] + «c:1»
+«:julia-ts-macro-face:@m» [«c:1», «c:2»] + «c:1»
+
+«x:# ==============================»
+«x:# quote expressions»
+«x:# ==============================»
+
+«:julia-ts-quoted-symbol-face::foo»
+«:julia-ts-quoted-symbol-face::const»
+
+«:julia-ts-quoted-symbol-face::(x; y)»
+«:julia-ts-quoted-symbol-face::(x, y)»
+«:julia-ts-quoted-symbol-face::[x, y, z]»
+
+«:julia-ts-quoted-symbol-face::+»
+«:julia-ts-quoted-symbol-face::->»
+«:julia-ts-quoted-symbol-face::(::)»
+
+«x:# ==============================»
+«x:# interpolation expressions»
+«x:# ==============================»
+
+«:julia-ts-interpolation-expression-face:$»«D:foo»
+«:julia-ts-interpolation-expression-face:$»«D:obj».field
+«:julia-ts-interpolation-expression-face:$(»«D:obj.field»«:julia-ts-interpolation-expression-face:)»
+«:julia-ts-interpolation-expression-face:$»«D:f»(x)
+«:julia-ts-interpolation-expression-face:$»«D:f»[«c:1», «c:2»]
+$«s:"foo"»
+
+«k:using» «:julia-ts-interpolation-expression-face:$»«D:Package»: «:julia-ts-interpolation-expression-face:$(»«D:name»«:julia-ts-interpolation-expression-face:)»
+
+«x:# Similar definitions in Gadfly/src/varset.jl»
+«k:mutable» «k:struct» «:julia-ts-interpolation-expression-face:$(»«D:name»«:julia-ts-interpolation-expression-face:)»
+  «:julia-ts-interpolation-expression-face:$(»«D:vars...»«:julia-ts-interpolation-expression-face:)»
```

Comment:
Should all these parentheses be default rather than `julia-ts-interpolation-expression-face`. Prefacing them with a `$` doesn't actually change their semantics in julia afaict.

Reply:
My reasoning here was this: the interpolated expression is wrapped in `$( ... )`. The parentheses are required because of operator precedence. Inside the interpolated expression, there are often other pairs of parentheses. Visually, it helps a lot if the start and end of the interpolation expression stand out. This fontification achieves that, they definitely help me a lot. A bit similarly to how `:( ... )` gets fontified as `quoted-symbol-face` (albeit together with the wrapped content).